packages/react/src/domain/handles.ts

```diff
@@ -29,6 +30,10 @@ const SSO_FACTORS = ["oidc", "saml"];
 function getPossibleHandleTypes(factor: Factor): Set<HandleType> {
   const handleTypes = new Set<HandleType>();
 
+  if (isFactorWithAllowedHandleTypes(factor)) {
```

Comment:
This order of checks allows a weird situation where you specify the extra option to use a specific handle type before checking if the given factor actually requires that handle type.

To be on the safe side, I would turn this around so that you still check if the factor actually requires the handle type specified in the `allowedHandleTypes` thing and only add it in these cases:
- there is no `allowedHandleTypes` prop present
- `allowedHandleTypes` is present and the given type is included